### ISSUE TITLE
Simple RateRule model and sensitivities

### ISSUE THREAD

**Issue by lcontento** [AMICI-dev/AMICI#1230]
A simple SBML model with one species `z`, one parameter `alpha` and `zdot = alpha` set by rate rule crashes AMICI when I try to compute state sensitivities (if sensitivity is not requested it works). MWE is
```python
with open('test.xml', 'w') as f:
    f.write("""\
<?xml version="1.0" encoding="UTF-8"?>
<sbml xmlns="http://www.sbml.org/sbml/level2/version5" level="2" version="5">
    <model id="spline_test">
        <listOfCompartments>
            <compartment id="compartment" size="1"/>
        </listOfCompartments>
        <listOfSpecies>
            <species id="z" compartment="compartment" initialAmount="0"/>
        </listOfSpecies>
        <listOfParameters>
            <parameter id="alpha" value="1.0"/>
        </listOfParameters>
        <listOfRules>
            <rateRule variable="z">
                <math xmlns="http://www.w3.org/1998/Math/MathML">
                    <ci> alpha </ci>
                </math>
            </rateRule>
        </listOfRules>
    </model>
</sbml>""")

from amici.sbml_import import SbmlImporter
imp = SbmlImporter("test.xml")
imp.sbml2amici("test", "amici_models", compile=True)

import sys
sys.path.insert(0, "amici_models")
import test
model = test.getModel()

import amici
solver = model.getSolver()
solver.setRelativeTolerance(1e-6)
solver.setAbsoluteTolerance(1e-9)
solver.setSensitivityOrder(amici.SensitivityOrder_first)
solver.setSensitivityMethod(amici.SensitivityMethod_forward)

import numpy as np
model.setTimepoints(np.linspace(0, 1, 10))

amici.runAmiciSimulation(model, solver)
```
Is it a bug in AMICI? Or am I doing something wrong?

**Comment by dweindl** [comment]
By crash you mean segfault? No warnings/errors printed?

**Comment by lcontento** [comment]
> By crash you mean segfault? No warnings/errors printed?

Yes.

**Comment by yannikschaelte** [comment]
The parameter name "z" is no problem; the return has `ret['y']` filled. How does AMICI deal with no obervable specifications?

**Comment by dweindl** [comment]
Confirmed.

Sparse matrix matrix multiplication crashes at 
```
amici::SUNMatrixWrapper::multiply (this=0x27407c8, c=..., b=...) at amici/src/sundials_matrix_wrapper.cpp:222
222                     for (sunindextype k = indexptrs_ptr_[i]; k < indexptrs_ptr_[i + 1];
```

with `i = 1`.

**Comment by dweindl** [comment]
@lcontento : Should be fixed on `fix_1230`.

**Comment by dweindl** [comment]
... and: great minimal reproducible example - appreciated.

**Comment by lcontento** [comment]
> @lcontento : Should be fixed on ~`fix_1230`~ `develop`.

Great, thanks for the fast fix! Now I can run it without segmentation faults.
However, the sensitivities `rdata['sx']` are all zeros...

**Comment by dweindl** [comment]
Seems like the symbols for the derivatives are not generated correctly for rate rules...
I think `ODEModel.symbols['species']['dt']` is ignored there.
Could you look into that, @dilpath ?

**Comment by dweindl** [comment]
@FFroehlich : What about adding gradient checks for the SBML test suite?

**Comment by dweindl** [comment]
> Seems like the symbols for the derivatives are not generated correctly for rate rules...
> I think `ODEModel.symbols['species']['dt']` is ignored there.
> Could you look into that, @dilpath ?

As far as I can see `dxdotdp_explicit` ignores parameters in `p`, assuming they are handled via `w` and respective derivatives. So one would have to add `ODEModel.symbols['species']['dt']` to `w`, which is currently empty for this model.

**Comment by lcontento** [comment]
The problem lies with the "stripped_symbols" (not sure what they are) that are used to compute the `dxdotdp` derivative.
Such symbols are generated as `Symbol(SBML_parameter_name)", so that there are two problems:
* if the SBML parameter "name" attribute is not set (which is not required by the specification) then the stripped symbol will be `None`
* even by setting the "name" attribute in the SBML file, the resulting symbol will have no "real=True" assumption so SymPy will think that it's different than the symbol inside `symbols['species']['dt']`, leading to the parameter derivative being zero.

**Comment by lcontento** [comment]
Looking at the comments, it seems to me that stripped symbols are only important for PySB problems and should not be relevant for SBML models. Would making `ODEModel._generate_symbol()` always return the non-stripped version when we are loading from SBML be a reasonable fix?

**Comment by lcontento** [comment]
As for the problem with "name" not being defined in the SBML file, would it be OK to just default it to the "id" attribute when it is not present?

**Comment by FFroehlich** [comment]
> @FFroehlich : What about adding gradient checks for the SBML test suite?

Yes thats something we should do. Already struggling to tweak FD computations for the pysb examples though.

**Comment by lcontento** [comment]
Fixed by #1231 and #1235.

**Comment by FFroehlich** [comment]
not on master yet
